lib/msal-node/src/client/ClientAssertion.ts

```diff
@@ -21,7 +21,8 @@ import { JwtConstants } from "../utils/Constants.js";
 export class ClientAssertion {
     private jwt: string;
     private privateKey: string;
-    private thumbprint: string;
+    private thumbprint: string | undefined;
```

Comment:
Still confused about `string | undefined`. Shouldn't you use `string` and let the `null or empty` value denote the existance of a value? Feels like this is poluting the public API.

Reply:
addressed this with code improvement